src/components/Garden/Preferences/EVMExecutor.js

```diff
@@ -0,0 +1,207 @@
+
+import MultiModal from '@components/MultiModal/MultiModal'
+import CreateDecisionScreens from '../ModalFlows/CreateDecisionScreens/CreateDecisionScreens'
+
+import { useGardens } from '@/providers/Gardens'
+import { useWallet } from '@/providers/Wallet'
+
+import { SHORTENED_APPS_NAMES } from '@utils/app-utils'
+// import { asciiToHex } from '@utils/web3-utils'
+
+import actions from '../../../actions/garden-action-types'
+import radspec from '../../../radspec'
+
+function EVMExecutor() {
+  const { account, ethers } = useWallet()
+  const { connectedGarden } = useGardens()
+  const [createDecisionModalVisible, setCreateDecisionModalVisible] = useState(
+    false
+  )
+  const [evmcrispr, setEvmcrispr] = useState(null)
+  const [installedApps, setInstalledApps] = useState([])
+  const [selectedApp, setSelectedApp] = useState(null)
+  const [selectedFunction, setSelectedFunction] = useState(null)
+  const [parameters, setParameters] = useState([])
+
+  useEffect(() => {
+    async function getEvmCrispr() {
+      if (!connectedGarden) {
+        return
+      }
+      const crispr = await EVMcrispr.create(
+        connectedGarden.address,
+        ethers.getSigner()
+      )
+
+      setEvmcrispr(crispr)
+    }
+    getEvmCrispr()
+  }, [connectedGarden, ethers])
+
+  useEffect(() => {
+    if (!evmcrispr) {
+      return
+    }
+    const apps = evmcrispr.apps()
+    setInstalledApps(apps)
+  }, [evmcrispr])
+
+  const shortenedAppsNames = installedApps.map(appName => {
+    const dotIndex = appName.indexOf('.')
+    return (
+      SHORTENED_APPS_NAMES.get(
+        dotIndex > 0
+          ? appName.substr(0, appName.indexOf('.'))
+          : appName.slice(0, -2)
+      ) || appName.slice(0, -2)
+    )
+  })
+
+  const functionList = useMemo(() => {
+    if (!evmcrispr || shortenedAppsNames?.length === 0 || !selectedApp) {
+      return
+    }
+
+    const appName = installedApps[selectedApp]
+
+    const appFunctions = Object.getOwnPropertyNames(evmcrispr.call(appName))
+    return appFunctions
+  }, [evmcrispr, selectedApp, installedApps, shortenedAppsNames])
+
+  const requiredParameters = useMemo(() => {
+    if (!evmcrispr || !selectedApp || !selectedFunction) {
+      return
+    }
+
+    const { paramNames, paramTypes } = evmcrispr.call(
+      installedApps[selectedApp]
+    )[functionList[selectedFunction]]
+
+    return paramNames.map((parameter, index) => {
+      return [parameter, paramTypes[index]]
+    })
+  }, [evmcrispr, installedApps, functionList, selectedApp, selectedFunction])
+
+  const handleOnChangeParameters = useCallback((index, event) => {
+    const newValue = event.target.value
+    setParameters(prevState => {
+      const newArray = [...prevState]
+      newArray[index] = newValue
+      return newArray
+    })
+  }, [])
+
+  const handleOnCreateIntent = useCallback(async () => {
+    const description = radspec[actions.NEW_DECISION]()
+    const type = actions.NEW_DECISION
```

Comment:
Let´s perhaps add a TODO: here to move this to the file of `useActions` as i think we should have them all there once we change the rest to use the `evmscripr` (if we decide to do so)